x-pack/plugins/security_solution/public/detection_engine/rule_management/api/hooks/use_bulk_action_mutation.ts

```diff
@@ -18,40 +18,40 @@ import { DETECTION_ENGINE_RULES_BULK_ACTION } from '../../../../../common/consta
 export const BULK_ACTION_MUTATION_KEY = ['POST', DETECTION_ENGINE_RULES_BULK_ACTION];
 
 export const useBulkActionMutation = (
-  options?: UseMutationOptions<BulkActionResponse, Error, BulkActionProps>
+  options?: UseMutationOptions<BulkActionResponse, Error, BulkAction>
 ) => {
   const invalidateFindRulesQuery = useInvalidateFindRulesQuery();
   const invalidateFetchRuleByIdQuery = useInvalidateFetchRuleByIdQuery();
   const invalidateFetchTagsQuery = useInvalidateFetchTagsQuery();
   const invalidateFetchPrebuiltRulesStatusQuery = useInvalidateFetchPrebuiltRulesStatusQuery();
   const updateRulesCache = useUpdateRulesCache();
 
-  return useMutation<BulkActionResponse, Error, BulkActionProps>(
-    (action: BulkActionProps) => performBulkAction(action),
+  return useMutation<BulkActionResponse, Error, BulkAction>(
+    (bulkActionDescriptor: BulkAction) => performBulkAction(bulkActionDescriptor),
```

Comment:
@maximpn I think `bulkActionDescriptor: BulkAction` looks less clear than `action: BulkActionProps`, I'm not sure what the word "descriptor" is supposed to mean in this case. Suffixes like `Props`, `Args` and `Params` are quite common in the security_solution codebase - it's already three of them, let's not introduce yet another one :)

Also, it's not a bulk action itself, it's rather the arguments for it that the user provides, so `BulkActionProps` is also more precise than `BulkAction` in this case.

Finally, please be mindful of renamings and refactorings. Many of the functions exported from `security_solution/public/detection_engine/rule_management/api/api.ts` accept some kind of `Props`, for example `FetchRuleProps`. Moving away from this naming convention in one function out of many doesn't bring value. If the naming could be improved it should be done in a separate PR consistently for all the functions.

My suggestion would be to revert the renaming back.

Reply:
Thanks for the detailed explanation. I've changed the input type of `performBulkAction` on `PerformBulkActionProps`.